scripts/o3de/o3de/engine_template.py

```diff
@@ -2280,69 +2298,62 @@ def create_gem(gem_path: pathlib.Path,
-                    s.write(json.dumps(restricted_json_data, indent=4) + '\n')
+                with open(restricted_json, 'r') as s:
+                    try:
+                        restricted_json_data = json.load(s)
+                    except json.JSONDecodeError as e:
+                        logger.error(f'Failed to load restricted json {restricted_json}.')
+                        return 1
 
-            with open(restricted_json, 'r') as s:
                 try:
-                    restricted_json_data = json.load(s)
-                except json.JSONDecodeError as e:
-                    logger.error(f'Failed to load restricted json {restricted_json}.')
+                    restricted_name = restricted_json_data["restricted_name"]
+                except KeyError as e:
+                    logger.error(f'Failed to read "restricted_name" from restricted json {restricted_json}.')
                     return 1
 
-            try:
-                restricted_name = restricted_json_data["restricted_name"]
-            except KeyError as e:
-                logger.error(f'Failed to read "restricted_name" from restricted json {restricted_json}.')
-                return 1
-
-            # set the "restricted_name": <restricted_name> element of the gem.json
-            gem_json = gem_path / 'gem.json'
-            if not validation.valid_o3de_gem_json(gem_json):
-                logger.error(f'Gem json {gem_json} is not valid.')
-                return 1
-
-            with open(gem_json, 'r') as s:
-                try:
-                    gem_json_data = json.load(s)
-                except json.JSONDecodeError as e:
-                    logger.error(f'Failed to load gem json {gem_json}.')
+                # set the "restricted_name": <restricted_name> element of the gem.json
+                gem_json = gem_path / 'gem.json'
+                if not validation.valid_o3de_gem_json(gem_json):
+                    logger.error(f'Gem json {gem_json} is not valid.')
                     return 1
 
-            gem_json_data.update({"restricted": restricted_name})
-            os.unlink(gem_json)
-            with open(gem_json, 'w') as s:
-                try:
-                    s.write(json.dumps(gem_json_data, indent=4) + '\n')
-                except OSError as e:
-                    logger.error(f'Failed to write project json {gem_json}.')
-                    return 1
-            '''
-            for restricted_platform in restricted_platforms:
-                restricted_gem = gem_restricted_path / restricted_platform / gem_name
-                os.makedirs(restricted_gem, exist_ok=True)
-                cmakelists_file_name = restricted_gem / 'CMakeLists.txt'
-                if not os.path.isfile(cmakelists_file_name):
-                    with open(cmakelists_file_name, 'w') as d:
-                        if keep_license_text:
-                            d.write(O3DE_LICENSE_TEXT)
-            '''
```

Comment:
Note that I removed this because its a triple-backtick raw string commented block that does nothing, in python, the equivalent of commented out code.